Common/src/grid_movement/CSurfaceMovement.cpp

```diff
@@ -353,6 +359,144 @@ void CSurfaceMovement::SetSurface_Deformation(CGeometry *geometry, CConfig *conf
+
+                if (rank == MASTER_NODE) cout << "Checking FFD box intersections with the solid surfaces." << endl;
+                  CheckFFDIntersections(geometry, config, FFDBox[iFFDBox], iFFDBox);
+
+                /*--- Compute the parametric coordinates of the child box
+                control points (using the parent FFDBox)  ---*/
+                for (iChild = 0; iChild < FFDBox[iFFDBox]->GetnChildFFDBox(); iChild++) {
+                  FFDBoxTag = FFDBox[iFFDBox]->GetChildFFDBoxTag(iChild);
+                  for (jFFDBox = 0; jFFDBox < GetnFFDBox(); jFFDBox++)
+                    if (FFDBoxTag == FFDBox[jFFDBox]->GetTag()) break;
+                    SetParametricCoordCP(geometry, config, FFDBox[iFFDBox], FFDBox[jFFDBox]);
+                }
+
+                /*--- Update the parametric coordinates if it is a child FFDBox ---*/
+                iLevel = FFDBox[iFFDBox]->GetLevel();
+                if (iLevel > 0) UpdateParametricCoord(geometry, config, FFDBox[iFFDBox], iFFDBox);
+
+                /*--- Apply the design variables to the control point position ---*/
+
+                for (iDV = 0; iDV < config->GetnDV(); iDV++) {
+                  switch ( config->GetDesign_Variable(iDV) ) {
+                    case FFD_CONTROL_POINT_2D : SetFFDCPChange_2D(geometry, config, FFDBox[iFFDBox], FFDBox, iDV, false); break;
+                    case FFD_CAMBER_2D :        SetFFDCamber_2D(geometry, config, FFDBox[iFFDBox], FFDBox, iDV, false); break;
+                    case FFD_THICKNESS_2D :     SetFFDThickness_2D(geometry, config, FFDBox[iFFDBox], FFDBox, iDV, false); break;
+                    case FFD_TWIST_2D :         SetFFDTwist_2D(geometry, config, FFDBox[iFFDBox], FFDBox, iDV, false); break;
+                    case FFD_CONTROL_POINT :    SetFFDCPChange(geometry, config, FFDBox[iFFDBox], FFDBox, iDV, false); break;
+                    case FFD_NACELLE :          SetFFDNacelle(geometry, config, FFDBox[iFFDBox], FFDBox, iDV, false); break;
+                    case FFD_GULL :             SetFFDGull(geometry, config, FFDBox[iFFDBox], FFDBox, iDV, false); break;
+                    case FFD_TWIST :            SetFFDTwist(geometry, config, FFDBox[iFFDBox], FFDBox, iDV, false); break;
+                    case FFD_ROTATION :         SetFFDRotation(geometry, config, FFDBox[iFFDBox], FFDBox, iDV, false); break;
+                    case FFD_CONTROL_SURFACE :  SetFFDControl_Surface(geometry, config, FFDBox[iFFDBox], FFDBox, iDV, false); break;
+                    case FFD_CAMBER :           SetFFDCamber(geometry, config, FFDBox[iFFDBox], FFDBox, iDV, false); break;
+                    case FFD_THICKNESS :        SetFFDThickness(geometry, config, FFDBox[iFFDBox], FFDBox, iDV, false); break;
+                    case FFD_ANGLE_OF_ATTACK :  SetFFDAngleOfAttack(geometry, config, FFDBox[iFFDBox], FFDBox, iDV, false); break;
+                  }
+                }
```

Comment:
This can also probably become a function and re-used in other areas of the code.